presto-common/src/main/java/com/facebook/presto/common/block/AbstractMapBlock.java

```diff
@@ -194,6 +194,22 @@ public long getRegionLogicalSizeInBytes(int position, int length)
                 Integer.BYTES * HASH_MULTIPLIER * entryCount;
     }
 
+    @Override
+    public long getApproximateRegionLogicalSizeInBytes(int position, int length)
+    {
+        int positionCount = getPositionCount();
+        checkValidRegion(positionCount, position, length);
+
+        int entriesStart = getOffset(position);
+        int entriesEnd = getOffset(position + length);
+        int entryCount = entriesEnd - entriesStart;
+
+        return getRawKeyBlock().getApproximateRegionLogicalSizeInBytes(entriesStart, entryCount) +
+                getRawValueBlock().getApproximateRegionLogicalSizeInBytes(entriesStart, entryCount) +
+                (Integer.BYTES + Byte.BYTES) * positionCount +
```

Comment:
positionCount -> entryCount or length?

Reply:
@mbasmanova thanks for reading so carefully! It shoudl be length. I also added the comments
```
                (Integer.BYTES + Byte.BYTES) * length +         // offsets and mapIsNull
                Integer.BYTES * HASH_MULTIPLIER * entryCount;   // hashtables
```

Reply:
@yingsu00 Thanks for adding comments. These make it very easy to understand what the code is doing and why.